src/graph/graph.c

```diff
@@ -609,79 +591,45 @@ DataBlockIterator *Graph_ScanEdges(const Graph *g) {
-        g->_labels = rm_realloc(g->_labels, g->label_cap * sizeof(GrB_Matrix));
-    }
-
-    GrB_Matrix_new(&g->_labels[g->label_count++], GrB_BOOL, _Graph_NodeCap(g), _Graph_NodeCap(g));
+    // The array capacity is guaranteed by the GraphContext calling function.
+    GrB_Matrix_new(&g->labels[g->label_count++], GrB_BOOL, _Graph_NodeCap(g), _Graph_NodeCap(g));
     return g->label_count-1;
 }
 
-int Graph_AddRelation(Graph *g) {
+int Graph_AddRelationType(Graph *g) {
     assert(g);
-
-    // Make sure we've got room for a new relation matrix.
-    if(g->relation_count == g->relation_cap) {
-        g->relation_cap += 4;   // allocate room for 4 new matrices.
-        g->_relations = rm_realloc(g->_relations, g->relation_cap * sizeof(GrB_Matrix));
-    }
-
-    GrB_Matrix_new(&g->_relations[g->relation_count++], GrB_BOOL, _Graph_NodeCap(g), _Graph_NodeCap(g));
+    // The array capacity is guaranteed by the GraphContext calling function.
+    GrB_Matrix_new(&g->relations[g->relation_count++], GrB_BOOL, _Graph_NodeCap(g), _Graph_NodeCap(g));
     return g->relation_count-1;
 }
 
-GrB_Matrix Graph_GetAdjacencyMatrix(const Graph *g) {
+GrB_Matrix Graph_GetAdjacencyMatrix(const Graph *g, bool locked) {
     assert(g);
     GrB_Matrix m = g->adjacency_matrix;
-    _Graph_SynchronizeMatrix(g, m);
+    _Graph_SynchronizeMatrix(g, m, locked);
     return m;
 }
 
-GrB_Matrix Graph_GetLabel(const Graph *g, int label_idx) {
+GrB_Matrix Graph_GetLabel(const Graph *g, int label_idx, bool locked) {
     assert(g && label_idx < g->label_count);
-    GrB_Matrix m = g->_labels[label_idx];
-    _Graph_SynchronizeMatrix(g, m);
+    GrB_Matrix m = g->labels[label_idx];
+    _Graph_SynchronizeMatrix(g, m, locked);
     return m;
 }
 
-GrB_Matrix Graph_GetRelation(const Graph *g, int relation_idx) {
+GrB_Matrix Graph_GetRelation(const Graph *g, int relation_idx, bool locked) {
     assert(g && (relation_idx == GRAPH_NO_RELATION || relation_idx < g->relation_count));
     GrB_Matrix m;
 
     if(relation_idx == GRAPH_NO_RELATION) {
-        m = Graph_GetAdjacencyMatrix(g);
+        m = Graph_GetAdjacencyMatrix(g, locked);
     } else {
-        m = g->_relations[relation_idx];
-        _Graph_SynchronizeMatrix(g, m);
+        m = g->relations[relation_idx];
+        _Graph_SynchronizeMatrix(g, m, locked);
     }
     return m;
 }
 
-void Graph_CommitPendingOps(Graph *g) {
-    /* GraphBLAS might delay execution of operations to a later stage,
-     * here we're forcing GraphBLAS to execute all of its pending operations
-     * by asking for the number of entries in each matrix. */
-
-    GrB_Matrix M;
-    GrB_Index nvals;
-
-    M = Graph_GetAdjacencyMatrix(g);
-    GrB_Matrix_nvals(&nvals, M);
-
-    for(int i = 0; i < g->relation_count; i++) {
-        M = Graph_GetRelation(g, i);
-        GrB_Matrix_nvals(&nvals, M);
-    }
-
-    for(int i = 0; i < g->label_count; i++) {
-        M = Graph_GetLabel(g, i);
-        GrB_Matrix_nvals(&nvals, M);
-    }
-}
```

Comment:
This function is unused now that we stopped eager synchronization in bulk insertion.